logstash-core/src/main/java/org/logstash/plugins/factory/ExecutionContext.java

```diff
@@ -0,0 +1,77 @@
+package org.logstash.plugins.factory;
+
+import co.elastic.logstash.api.Context;
+import co.elastic.logstash.api.DeadLetterQueueWriter;
+import org.jruby.Ruby;
+import org.jruby.RubyBasicObject;
+import org.jruby.RubyClass;
+import org.jruby.anno.JRubyClass;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.RubyUtil;
+import org.logstash.common.AbstractDeadLetterQueueWriterExt;
+import org.logstash.common.DLQWriterAdapter;
+import org.logstash.common.NullDeadLetterQueueWriter;
+import org.logstash.execution.ExecutionContextExt;
+import org.logstash.instrument.metrics.AbstractNamespacedMetricExt;
+import org.logstash.plugins.ContextImpl;
+import org.logstash.plugins.NamespacedMetricImpl;
+import org.logstash.plugins.PluginLookup;
+
+@JRubyClass(name = "ExecutionContextFactory")
+public final class ExecutionContext extends RubyBasicObject {
+
+    private static final long serialVersionUID = 1L;
+
+    private IRubyObject agent;
+
+    private IRubyObject pipeline;
+
+    private IRubyObject dlqWriter;
+
+    public ExecutionContext(final Ruby runtime, final RubyClass metaClass) {
+        super(runtime, metaClass);
+    }
+
+    @JRubyMethod
+    public ExecutionContext initialize(final ThreadContext context,
+                                       final IRubyObject agent, final IRubyObject pipeline, final IRubyObject dlqWriter) {
+        this.agent = agent;
+        this.pipeline = pipeline;
+        this.dlqWriter = dlqWriter;
+        return this;
+    }
+
+    @JRubyMethod
+    public ExecutionContextExt create(final ThreadContext context, final IRubyObject id,
```

Comment:
:thinking: a bit confusing: `ExecutionContext` class is a factory for the `ExcutionContextExt` type ...
is it justified that we're not following the `#{RubyName}Ext` convention here?